### PR TITLE
[SYCL] Fix unused variable errors on post-commit for forward_progress extension

### PR DESCRIPTION
**Pull request by lbushi25**
The merge of https://github.com/intel/llvm/pull/13389 has caused post-commit failures due to three unused variable warnings. 
This PR fixes that. One of these warnings also revealed a bug in my original PR.

### CODE REVIEW

**Review by aelovikov-intel — APPROVED** [review]

**Comment by lbushi25** [comment]
@intel/llvm-gatekeepers ,
The failures are unrelated to my changes.

**Comment by aelovikov-intel** [comment]
> @intel/llvm-gatekeepers , The failures are unrelated to my changes.

Testing hasn't finished yet:

image

Please don't tag until it's complete.

**Comment by aelovikov-intel** [comment]
> @intel/llvm-gatekeepers , The failures are unrelated to my changes.

What other explanation do you have for them? Your change is the last in post-commit queue and everything was green before it.

**Comment by lbushi25** [comment]
> > @intel/llvm-gatekeepers , The failures are unrelated to my changes.
> 
> What other explanation do you have for them? Your change is the last in post-commit queue and everything was green before it.

I doubt that my original PR is causing them to fail because there has been another PR created( https://github.com/intel/llvm/pull/13794) after that was merged and its Pre Commit on Linux is passing. I'll try a rerun of the failing task.

**Comment by lbushi25** [comment]
The failure is persisting even after a rerun but I cannot reproduce it on my local machine. Will look into this more!

**Comment by steffenlarsen** [comment]
@lbushi25 was right, the failures are completely unrelated and are not very obvious. https://github.com/intel/llvm/pull/13801 fixes them and should hopefully help us prevent similar problems in the unittests in the future.

File: sycl/source/detail/device_impl.cpp [hunk 1]
```diff
@@ -877,6 +879,7 @@ device_impl::getProgressGuarantee(
       ext::oneapi::experimental::forward_progress_guarantee;
   using execution_scope = ext::oneapi::experimental::execution_scope;
   const int executionScopeSize = 4;
+  (void)coordinationScope;
```
[the remark this answers is not in the document]
Comment (steffenlarsen) [reply sycl/source/detail/device_impl.cpp:880]: I believe it is a bug that this isn't used and that a patch is incoming to fix the use, so let's keep it like this.

**Pull request merged**